inst/templates/tidy_contributing.md

```diff
@@ -0,0 +1,147 @@
+# Contributing to {{{package}}}
+
+-   [Prerequisites](#prerequisites)
+-   [Package mechanics](#package-mechanics)
+    -   [Fork, clone, branch](#fork-clone-branch)
+    -   [Check](#check)
+    -   [Style](#style)
+    -   [Document](#document)
+    -   [Test](#test)
+    -   [NEWS](#news)
+    -   [Re-check](#re-check)
+    -   [Commit](#commit)
+-   [Push and pull](#push-and-pull)
+    -   [Review, revise, repeat](#review-revise-repeat)
+-   [Resources](#resources)
+-   [Code of Conduct](#code-of-conduct)
+
+This explains how to propose a change to {{{package}}} via a pull request using
+Git and Github. 
+
+For more general info about contributing to the tidyverse, see the 
+[Resources](#resources) at the end of this document.
+
+## Prerequisites
+
+Before you do a pull request, you should always file an issue and make sure
+someone from the tidyverse team agrees that it’s a problem, and is happy with
+your basic proposal for fixing it. If you’ve found a bug, first create a minimal
+[reprex](https://www.tidyverse.org/help/#reprex).
+
+## Package mechanics
+
+### Fork, clone, branch
+
+The first thing you'll need to do is to fork the 
+[{{{package}}} GitHub repo](https://github.com/tidyverse/{{{package}}}), and 
+then clone it locally. We recommend that you create a branch for each PR.
+
+### Check
+
+Before changing anything, make sure the package still passes `R CMD check`
+locally for you. You'll do this again _after_ you've made changes, but it's best
```

Comment:
You could wiggle a bit here. Hopefully the package does currently pass `R CMD check`! But we frequently go through periods of failure.

I guess the real baseline is "whatever the current Travis/AppVeyor results are". Possible re-wording:

> Before changing anything, make sure the package still passes `R CMD check` locally for you. When in doubt, compare your `R CMD check` results with current results for {{{package}}} on Travis (checks on Linux and/or MacOS) and, if applicable, AppVeyor (checks on Windows). You'll do this again before you finalize your pull request, but this baseline will make it easier to pinpoint any problems introduced by your changes.

where "Travis" and "AppVeyor" could probably be templated into the correct links.

Reply:
Ooh— that's great! @hadley Is there a naming convention I should use to make use of `travis_info` here? https://github.com/r-lib/usethis/blob/556e08a0d2f7a1341a5c1a5a3c36486023b34d02/R/ci.R#L31
Or `travis_info$url`?
https://github.com/r-lib/usethis/blob/556e08a0d2f7a1341a5c1a5a3c36486023b34d02/R/ci.R#L37

Reply:
You basically get to make up the convention when you write the `use_tidy_contributing()` function. I'd suggest using `travis_url`, then implementing that function in a separate PR once we're good here (since there is so much commentary already, and we're close to being done)